app/assets/javascripts/discourse/app/widgets/post-menu.js

```diff
@@ -477,27 +482,65 @@ export default createWidget("post-menu", {
       collapsed: true,
       likedUsers: [],
       readers: [],
-      adminVisible: false,
     };
   },
 
   buildKey: (attrs) => `post-menu-${attrs.id}`,
 
   attachButton(name) {
-    let buttonAtts = buildButton(name, this);
+    let buttonAttrs = buildButton(name, this);
+
+    if (buttonAttrs?.component) {
+      return [
+        new RenderGlimmer(
+          this,
+          buttonAttrs.tagName,
+          hbs`<@data.component
+            @permanentlyDeletePost={{@data.permanentlyDeletePost}}
+            @lockPost={{@data.lockPost}}
+            @unlockPost={{@data.unlockPost}}
+            @grantBadge={{@data.grantBadge}}
+            @rebakePost={{@data.rebakePost}}
+            @toggleWiki={{@data.toggleWiki}}
+            @changePostOwner={{@data.changePostOwner}}
+            @changeNotice={{@data.changeNotice}}
+            @togglePostType={{@data.togglePostType}}
+            @post={{@data.post}}
+            @transformedPost={{@data.transformedPost}}
+            @scheduleRerender={{@data.scheduleRerender}}
+          />`,
+          {
+            component: buttonAttrs.component,
+            transformedPost: this.attrs,
```

Comment:
`attrs` bit me the other day by having some weird side effects. We don't have access to `transformedPost` on `this` do we 🙏 ?